library/PreseedControl.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import json
+
+from library.common import *
+
+
+class PreseedControl(object):
+    """ This class handles the job of uploading a preseed file to MrP.
+        It shall only be called if there is a file to be uploaded, else you can
+        fetch the thing via the regular call in the Ansible role"""
+
+    def __init__(self, requester, preseed_name):
+        self.requester = requester
+        self.name = preseed_name
+
+    def _get_preseed_from_file(self, preseed_file, preseed_desc, preseed_type, public=False,
+                               knowngood=False):
+        json_preseed = {}
+        contents = ''
+
+        with open(preseed_file, 'r') as fd:
+            lines = fd.readlines()
+
+        for line in lines:
+            contents += line
+
+        json_preseed['content'] = contents
+        json_preseed['name'] = self.name
+        json_preseed['type'] = preseed_type
+        json_preseed['public'] = public == 'True'
+        json_preseed['known_good'] = knowngood == 'True'
+        if preseed_desc != '':
+            json_preseed['description'] = preseed_desc
+
+        return json_preseed
+
+    def check_for_existence(self):
+        url = '/api/v1/preseed?show_all=true'
+
+        try:
+            preseeds = self.requester.get(url)
+        except Exception as err:
+            raise err
+
+        for preseed in preseeds:
+            if preseed['name'] == self.name:
+                return preseed
+
+        return False
+
+    def get_preseed_id(self):
+        try:
+            preseed = self.check_for_existence()
+        except Exception as err:
+            raise err
+
+        if preseed != False:
+            return preseed['id']
```

Comment:
Same comment as above, return the dictionary and let the caller use what it needs